skimage/transform/_orientation.py

```diff
@@ -0,0 +1,179 @@
+        Indices of the components of `axis` that should be transformed.
+        If `None`, defaults to all of the indices of `axis`.
+
+    Returns
+    -------
+    rotation_matrix : (N, N) array
+        Orthogonal projection matrix.
+
+    References
+    ----------
+    .. [1] Ognyan Ivanov Zhelezov. One Modification which Increases Performance
+           of N-Dimensional Rotation Matrix Generation Algorithm. International
+           Journal of Chemistry, Mathematics, and Physics, Vol. 2 No. 2, 2018:
+           pp. 13-18. https://dx.doi.org/10.22161/ijcmp.2.2.1
+    .. [2] https://en.wikipedia.org/wiki/Rotation_matrix#In_three_dimensions
+
+
+    Examples
+    --------
+    >>> rotation_matrix = _axis_0_rotation_matrix([0, 1, 0])
+    >>> rotation_matrix @ [0, 1, 0]
+    array([ 1.,  0.,  0.])
+    """
+    ndim = len(unit_vector)
+
+    if indices is None:
+        indices = range(ndim)
+
+    rotation_matrix = np.eye(ndim)
+
+    # loop to create stages of 2D rotations around fixed axes
+    # that are multiplied to form our nD matrix; see: [2]_
+    for step in np.round(2 ** np.arange(np.log2(ndim))).astype(int):
+        plane_rotation_matrix = np.eye(ndim)
+
+        for n in range(0, ndim - step, step * 2):
+            if n + step >= len(indices):
+                break
+
+            # axes that make up this plane
+            i = indices[n]
+            j = indices[n + step]
+
+            # distance from origin in this plane
+            radius = np.hypot(unit_vector[i], unit_vector[j])
+
+            if radius > 0:
+                # calculation of coefficients
+                pcos = unit_vector[i] / radius
+                psin = -unit_vector[j] / radius
+
+                # base 2-dimensional rotation for this plane
+                plane_rotation_matrix[i, i] = pcos
+                plane_rotation_matrix[i, j] = -psin
+                plane_rotation_matrix[j, i] = psin
+                plane_rotation_matrix[j, j] = pcos
+
+                unit_vector[i] = radius
+                unit_vector[j] = 0
+
+        # compound current plane's rotation with previous ones'
+        rotation_matrix = plane_rotation_matrix @ rotation_matrix
+
+    return rotation_matrix
+
+
+def compute_rotation_matrix(src, dst, use_homogeneous_coords=False):
+    """Generate a matrix for the rotation of one vector to the direction
+    of another.
+
+    The MNMRG algorithm [1]_ can be summarized as follows:
+        1. directional vectors ``X`` and ``Y`` are normalized
+        2. a vector ``w`` is initialized containing the
+           indices of the differences between ``X`` and ``Y``
+        3. matrices ``Mx`` and ``My`` are generated for the rotation
+           of ``X`` and ``Y`` to the same axis for all indices
+           in ``w``
+        4. the inverse of ``My`` is mulitplied by ``Mx`` to form
+           the rotation matrix ``M`` which rotates vector ``X`` to the
+           direction of vector ``Y``
```

Comment:
Can we change all these to use active voice? "vectors are normalized" -> "normalize vectors"